README.md

```diff
@@ -87,25 +87,21 @@ from running `make build`, and then invoking it with the correct path under
 `./bazel-bin`. For example, if you are on a Linux machine, running `make build`
 will output a binary at `./bazel-bin/linux_amd64_stripped/cip`.
 
-# How it works
+# What it does
 
-At a high level, the promoter simply performs set operations ("set" as in
-mathematics, like in Venn diagrams). It first creates the set of all images in
-the destination registry (let's call it `D`). It also creates the set of all
-images defined in the promoter manifest as "promotion candidates in the
-manifest" (call it `M`).
+The promoter's behaviour can be described in terms of mathematical sets (as in Venn diagrams).
+Suppose `D` is the set of all images in the destination registry and `M` is the set of 
+images to be promoted (these are defined in the promoter manifest). Then:
 
-Given the above, we get the following results:
+- `M ∩ D` = images which do not need promoting since they are already present in the destination registry
+- `M \ D` = images that are copied
```

Comment:
Or, if we take into account those which cannot be found (in which case the definition of `S` needs moving too):
```suggestion
- `(M ∩ S) \ D` = images that are copied
```